From e35db37221daf0ba52087bda4c1975c220942a4e Mon Sep 17 00:00:00 2001
From: Thomas Preud'homme <robotux@celest.fr>
Date: Wed, 13 Jun 2012 19:34:57 +0200
Subject: Disable callsave_test test on arm

callsave_test uses alloca to perform its test but alloca is not
available on arm. Hence this test must be disabled.

Origin: vendor
Forwarded: not-needed
Last-Update: 2012-06-13
---
 tests/tcctest.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/tcctest.c b/tests/tcctest.c
index 1b04c1f..4ab4cc7 100644
--- a/tests/tcctest.c
+++ b/tests/tcctest.c
@@ -2665,6 +2665,8 @@ double get100 () { return 100.0; }
 
 void callsave_test(void)
 {
+/* alloca is not defined on arm */
+#if defined __i386__ || defined __x86_64__
   int i, s; double *d; double t;
   s = sizeof (double);
   printf ("callsavetest: %d\n", s);
@@ -2677,4 +2679,5 @@ void callsave_test(void)
      generates a segfault.  */
   i = d[0] > get100 ();
   printf ("%d\n", i);
+#endif
 }
